src/core_plugins/interpreter/public/functions/kibana.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export default () => ({
+  name: 'kibana',
+  type: 'kibana_context',
+  context: {},
+  help: i18n.translate('common.core_plugins.interpreter.public.functions.kibana.help', {
+    defaultMessage: 'Gets kibana global context'
+  }),
+  args: {},
+  fn(context, args, handlers) {
+    const initialContext = handlers.getInitialContext ? handlers.getInitialContext() : {};
+
+    if (context.query) {
+      const contextQueries = Array.isArray(context.query) ? context.query : [context.query];
+      initialContext.query = initialContext.query.concat(contextQueries);
+    }
+
+    if (context.filters) {
+      // merge filters
+      initialContext.filters = initialContext.filters.concat(context.filters);
+    }
+
+    const timeRange = initialContext.timeRange ? JSON.parse(initialContext.timeRange) : context.timeRange;
```

Comment:
That won't work, since `initialContext` will be a plain JSON object. `timeRange` is not in the form of a string, and does not need to be JSON parsed.